examples/feature_engineering_multiple_contexts/scenario_2/online_loader.py

```diff
@@ -0,0 +1,142 @@
+
+        :param client_id: id of the client to get data for.
+        :param features_needed: list of features to request from the feature store.
+        :return: a dataframe with the requested features.
+        """
+        # Example request:
+        # async with self.session.get(
+        #     f"{self.url}/v0/features",
+        #     params={
+        #         "client_id": client_id,
+        #         "features": features_needed,
+        #     },
+        # ) as resp:
+        #     data = await resp.json()
+        #     return pd.DataFrame(data["data"], index=[client_id])
+        # But to make this example work without you setting up a feature store we'll just return some fake data.
+        data = {
+            "reason_for_absence": 26,
+            "month_of_absence": 7,
+            "day_of_the_week": 3,
+            "seasons": 2,
+            "transportation_expense": 1,
+            "distance_from_residence_to_work": 1,
+            "service_time": 289,
+            "age": 36,
+            "work_load_average_per_day": 13,
+            "hit_target": 33,
+            "disciplinary_failure": 239.554,
+            "education": 97,
+            "son": 0,
+            "social_drinker": 1,
+            "social_smoker": 2,
+            "pet": 1,
+            "weight": 90,
+            "height": 172,
+            "body_mass_index": 30,  # remove this comma to make it valid json.
+        }
+        return pd.DataFrame(data, index=[client_id])
+
+    async def get_agg_features(self, entity_type: str) -> dict:
+        """Gets the aggregate features for the entity type."""
+        # Example request:
+        # async with self.session.get(
+        #     f"{self.url}/v0/agg_features",
+        #     params={
+        #         "entity_type": "client",
+        #     },
+        # ) as resp:
+        #     data = await resp.json()
+        #     return data
+        # But to make this example work without you setting up a feature store we'll just return some fake data.
+        return {"age_mean": 33.0, "age_std_dev": 13.0}
+
+
+# full set of available columns from the data source
+FEATURES_IN_STORE = [
+    "reason_for_absence",
+    "month_of_absence",
+    "day_of_the_week",
+    "seasons",
+    "transportation_expense",
+    "distance_from_residence_to_work",
+    "service_time",
+    "age",
+    "work_load_average_per_day",
+    "hit_target",
+    "disciplinary_failure",
+    "education",
+    "son",
+    "social_drinker",
+    "social_smoker",
+    "pet",
+    "weight",
+    "height",
+    "body_mass_index",
+]
+
+
+@tag(source="feature_store")
+@extract_columns(*FEATURES_IN_STORE)
+async def raw_data(client_id: str, feature_client: FeatureStoreHttpClient) -> pd.DataFrame:
+    """Extracts the raw data, renames the columns to be valid python variable names, and assigns an index.
+
+    Note: we could get cleverer here and only request the features we need. We can use the Hamilton Driver to ask
```

Comment:
Side-note: this feels like a new use-case that's akin to the `select FOO` issue. 
1. Load up data from a db
2. extract columns
3. just use a subset

Should be able to be:
1. Load just the data you need
2. extract all the columns 
3. Use a subset

Is there a recipe here? can't think of an easy implementation to make this natural, but worth thinking through...

Reply:
Maybe if the function doing the `select` knew the entire path, it could be smart about it... We could automatically pass that in as an input and allow the data loader piece to handle it?

Reply:
yep, it's the same deal. Not in scope for this scenario.